PySDM/backends/numba/impl/_algorithmic_methods.py

```diff
@@ -134,15 +134,71 @@ def coalescence_body(n, idx, length, attributes, gamma, healthy, is_first_in_pai
                     attributes[a, k] = attributes[a, j]
             if n[k] == 0 or n[j] == 0:
                 healthy[0] = 0
-
+                
     @staticmethod
     def coalescence(n, idx, length, attributes, gamma, healthy, is_first_in_pair):
         AlgorithmicMethods.coalescence_body(n.data, idx.data, length,
                                             attributes.data, gamma.data, healthy.data, is_first_in_pair.indicator.data)
+'''
+    
         
-    ## TODO: Emily implement breakup
     @staticmethod
     @numba.njit(**conf.JIT_FLAGS)
+    def collision_body(n, idx, length, attributes, gamma, rand, dyn, Ec, Eb, n_fragment, healthy, is_first_in_pair):
```

Comment:
We should think carefully about how a collision step calls breakup or coalescence code. We are doing a lot of extra initialization and randomization for the process that doesn't end up happening. It would be best if we could only do this work when we have to, and we could probably keep the modularization of dynamics that we've already built.